include/acquire.zarr.h

```diff
@@ -48,14 +63,26 @@ extern "C"
      */
     const char* Zarr_get_status_message(ZarrStatusCode status);
 
+    /**
+     * @brief Allocate memory for the dimension array in the Zarr stream settings struct.
+     * @param[in, out] settings The Zarr stream settings struct.
+     * @param dimension_count The number of dimensions in the dataset to allocate memory for.
+     * @return ZarrStatusCode_Success on success, or an error code on failure.
+     */
+    ZarrStatusCode ZarrStreamSettings_create_dimension_array(ZarrStreamSettings* settings, size_t dimension_count);
+
+    /**
+     * @brief Free memory for the dimension array in the Zarr stream settings struct.
+     * @param[in, out] settings The Zarr stream settings struct containing the dimension array to free.
+     */
+    void ZarrStreamSettings_destroy_dimension_array(ZarrStreamSettings* settings);
```

Comment:
It could, but there are two parameters that are set or reset (I double checked this and realized `dimension_count` is not actually getting reset in `ZarrStreamSettings_destroy_dimension_array` though, but I'll push a fix for that).